### PR TITLE
Updating CI build with AMD Roc images

### PR DESCRIPTION
**Pull request by dibryant**
FIxes for https://issues.redhat.com/browse/RHOAIENG-4608

Updated the CI build to include AMD Roc images

### CODE REVIEW

**Comment by atheo89** [comment]
Hey Diamond, to resolve the issue you need to run `make jobs` on your fork, this will create some new files that are needed to be commited. 

If you press on details button on the failed job, it redirects you on the logs page so you may see what it suggest you to fix the test.  https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/51945/pull-ci-openshift-release-master-generated-config/1791465803057467392

File: ci-operator/config/opendatahub-io/notebooks/opendatahub-io-notebooks-main.yaml [hunk 1]
```diff
@@ -169,6 +177,10 @@ images:
   dockerfile_path: Dockerfile
   from: cuda-c9s-python-3.9
   to: cuda-rstudio-c9s-python-3.9
+- context_dir: amd/c9s-python-3.9
```
Comment (atheo89) [inline ci-operator/config/opendatahub-io/notebooks/opendatahub-io-notebooks-main.yaml:180]: Shall we build also the minimal version `amd-jupyter-minimal-c9s-python-3.9`?

Comment (dibryant) [reply ci-operator/config/opendatahub-io/notebooks/opendatahub-io-notebooks-main.yaml:180]: Yes that was built lines 153-160

File: ci-operator/config/opendatahub-io/notebooks/opendatahub-io-notebooks-main.yaml [hunk 2]
```diff
@@ -150,6 +150,14 @@ images:
...
   from: intel-runtime-ml-ubi9-python-3.9
   to: jupyter-intel-ml-ubi9-python-3.9
+- context_dir: amd/ubi9-python-3.9
+  dockerfile_path: Dockerfile
+  from: base-ubi9-python-3.9
+  to: amd-ubi9-python-3.9
+- context_dir: jupyter/minimal/ubi9-python-3.9
+  dockerfile_path: Dockerfile
+  from: amd-ubi9-python-3.9
+  to: amd-jupyter-minimal-ubi9-python-3.9
```
Comment (harshad16) [inline ci-operator/config/opendatahub-io/notebooks/opendatahub-io-notebooks-main.yaml:160]: As we are not building UBI based image  in the first interaction, we can remove these.

File: ci-operator/config/opendatahub-io/notebooks/opendatahub-io-notebooks-main.yaml [hunk 3]
```diff
@@ -169,6 +177,10 @@ images:
   dockerfile_path: Dockerfile
   from: cuda-c9s-python-3.9
   to: cuda-rstudio-c9s-python-3.9
+- context_dir: amd/c9s-python-3.9
+  dockerfile_path: Dockerfile
+  from: base-c9s-python-3.9
+  to: amd-c9s-python-3.9
```
Comment (harshad16) [inline ci-operator/config/opendatahub-io/notebooks/opendatahub-io-notebooks-main.yaml:183]:
```suggestion
  to: amd-c9s-python-3.9
- context_dir: jupyter/minimal/ubi9-python-3.9
  dockerfile_path: Dockerfile
  from: amd-c9s-python-3.9
  to: amd-jupyter-minimal-c9s-python-3.9
```
based on the suggestion, we could actually us the minimal for testing

**Review by harshad16 — CHANGES_REQUESTED** [review]
Thanks for the work, 
some changes are needed

File: ci-operator/config/opendatahub-io/notebooks/opendatahub-io-notebooks-main.yaml [hunk 4]
```diff
@@ -169,6 +169,14 @@ images:
...
+- context_dir: jupyter/minimal/ubi9-python-3.9
```
Comment (atheo89) [inline ci-operator/config/opendatahub-io/notebooks/opendatahub-io-notebooks-main.yaml:176]: I am not sure if this will be build successfully

**Comment by atheo89** [comment]
/pj-rehearse pull-ci-opendatahub-io-notebooks-main-notebook-amd-c9s-python-3-9-pr-image-mirror
/pj-rehearse pull-ci-opendatahub-io-notebooks-main-notebook-amd-jupyter-minimal-c9s-python-3-9-pr-image-mirror

**Comment by atheo89** [comment]
/pj-rehearse pull-ci-opendatahub-io-notebooks-main-notebook-amd-jupyter-minimal-c9s-python-3-9-pr-image-mirror

**Comment by harshad16** [comment]
As this is merged: https://github.com/opendatahub-io/notebooks/pull/507
we can re-run the pj-rehearse

**Comment by harshad16** [comment]
/pj-rehearse pull-ci-opendatahub-io-notebooks-main-notebook-amd-c9s-python-3-9-pr-image-mirror

**Comment by harshad16** [comment]
/pj-rehearse pull-ci-opendatahub-io-notebooks-main-notebook-amd-jupyter-minimal-c9s-python-3-9-pr-image-mirror

**Comment by harshad16** [comment]
The rehearse seems to be working:
Screenshot from 2024-05-21 15-09-18

/pj-rehearse ack

**Pull request merged**